File: ShadowFrameworkLite20/src/shadow/renderer/data/transforms/SFTranslateFixed16Data.java
package shadow.renderer.data.transforms;

import shadow.geometry.data.SFFixedFloat16;
import shadow.math.SFMatrix3f;
import shadow.math.SFTransform3f;
import shadow.renderer.data.SFDataAsset;
import shadow.system.data.SFNamedParametersObject;
import shadow.system.data.objects.SFBinaryObject;

public class SFTranslateFixed16Data extends SFDataAsset<SFTransform3f>{

	private SFBinaryObject<SFFixedFloat16> x=new SFBinaryObject<SFFixedFloat16>(new SFFixedFloat16());
	private SFBinaryObject<SFFixedFloat16> y=new SFBinaryObject<SFFixedFloat16>(new SFFixedFloat16());
	private SFBinaryObject<SFFixedFloat16> z=new SFBinaryObject<SFFixedFloat16>(new SFFixedFloat16());
	
	public SFTranslateFixed16Data(){
		setup();
	}
	
	public SFTranslateFixed16Data(float x,float y,float z){
		setup();
		place(x, y, z);
	}

	public void setup() {
		SFNamedParametersObject parameters=new SFNamedParametersObject();
		parameters.addObject("x", x);
		parameters.addObject("y", y);
		parameters.addObject("z", z);
		setData(parameters);
	}
	
	public void place(float x,float y,float z){
		this.x.getBinaryValue().setFloat(x);
		this.y.getBinaryValue().setFloat(y);
		this.z.getBinaryValue().setFloat(z);
	}
	
	@Override
	protected SFTransform3f buildResource() {
		SFMatrix3f matrix=new SFMatrix3f();
		
		SFTransform3f transform=new SFTransform3f();
		transform.setMatrix(matrix);
		transform.setPosition(x.getBinaryValue().getFloat(),y.getBinaryValue().getFloat(),z.getBinaryValue().getFloat());
		
		return transform;
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/renderer/data/transforms/SFTranslateFixed16Data.js


function SFTranslateFixed16Data(){
	this.x=new SFBinaryObject(new SFFixedFloat16(0));
	this.y=new SFBinaryObject(new SFFixedFloat16(0));
	this.z=new SFBinaryObject(new SFFixedFloat16(0));
	var parameters=new SFNamedParametersObject();
	parameters.addObject(this.x);
	parameters.addObject(this.y);
	parameters.addObject(this.z);
	this.setData(parameters);
}

inherit(SFTranslateFixed16Data,SFDataAsset);


SFTranslateFixed16Data.prototype["buildResource"]=function(){
			var matrix=new SFMatrix3f();
			
			var transform=new SFTransform3f();
			transform.setMatrix(matrix);
			transform.setPosition(new SFVertex3f(this.x.getBinaryValue().getFloat(),
					this.y.getBinaryValue().getFloat(),
					this.z.getBinaryValue().getFloat()));
			
			return transform;
		};
		

SFTranslateFixed16Data.prototype["generateNewDatasetInstance"]=function(){
			return new SFTranslateFixed16Data();
		};
